internal/agent/upgrade.go

```diff
@@ -117,24 +121,51 @@ func Upgrade(
 
 	utils.SetEnv(c.Env)
 
-	args := []string{"upgrade", "--system.uri", fmt.Sprintf("docker:%s", img)}
-	args = append(args,
-		"--auth-username", authUser,
-		"--auth-password", authPass,
-		"--auth-server-address", authServer,
-		"--auth-type", authType,
-		"--auth-registry-token", registryToken,
-		"--auth-identity-token", identityToken,
-	)
-
+	// Load the upgrade Config from the system
+	resetConfig, err := elementalConfig.ReadConfigRun("/etc/elemental")
+	if err != nil {
+		return err
+	}
 	if debug {
-		fmt.Printf("Running command: 'elemental %s'", strings.Join(args, " "))
+		resetConfig.Logger.SetLevel(log.DebugLevel)
+	}
+
+	// Generate an auth object
+	auth := &types.AuthConfig{
+		Username:      authUser,
+		Password:      authPass,
+		ServerAddress: authServer,
+		Auth:          authType,
+		IdentityToken: identityToken,
+		RegistryToken: registryToken,
+	}
+
+	// Override the default luet to pass the auth
+	// Remember to create the temp dir!
+	tmpDir := elementalUtils.GetTempDir(&resetConfig.Config, "")
+	l := luet.NewLuet(luet.WithLogger(resetConfig.Logger), luet.WithAuth(auth), luet.WithLuetTempDir(tmpDir))
+	resetConfig.Luet = l
+
+	// Generate the upgrade spec
+	resetSpec, err := elementalConfig.ReadUpgradeSpec(resetConfig)
```

Comment:
This variable should probably be called `upgradeSpec`

Reply:
wops! good catch!